src/Voron/Debugging/StorageReportGenerator.cs

```diff
@@ -751,10 +752,8 @@ private static void GatherBalanceDistribution(Tree tree, TreePage page, Dictiona
         {
             if (page.IsLeaf)
             {
-                if (!histogram.TryGetValue(depth, out int value))
-                    value = 0;
-
-                histogram[depth] = value + 1;
+                ref var value = ref CollectionsMarshal.GetValueRefOrAddDefault(histogram, depth, out _);
```

Comment:
cc @arekpalinski

Reply:
it's the same effectively so everything good here